app/modules/enduser_setup.c

```diff
@@ -490,14 +555,26 @@ static int enduser_setup_http_load_payload(void)
   int file_len = (int) fs_tell(f);
   int err2 = fs_seek(f, 0, FS_SEEK_SET);
 
-  const char cl_hdr[] = "Content-length:%5d\r\n\r\n";
-  const size_t cl_len = LITLEN(cl_hdr) + 3; /* room to expand %4d */
+  char cl_hdr[30];
+  size_t ce_len = 0;
+  
+  c_sprintf(cl_hdr, http_header_content_len_fmt, file_len);
+  size_t cl_len = c_strlen(cl_hdr);
 
   if (f == 0 || err == -1 || err2 == -1)
   {
     ENDUSER_SETUP_DEBUG("enduser_setup_http_load_payload unable to load file enduser_setup.html, loading backup HTML.");
 
-    int payload_len = LITLEN(http_header_200) + cl_len + LITLEN(http_html_backup);
+    c_sprintf(cl_hdr, http_header_content_len_fmt, c_strlen(http_html_backup));
+    cl_len = c_strlen(cl_hdr);
+    
+    if (http_html_backup[0] == 0x1f && http_html_backup[1] == 0x8b)
```

Comment:
You're absolutely right. This was hand-merged, and I see in my fork that I used `sizeof` instead of `strlen`. Thanks for catching that!